src/driver-marsgaming/marsgaming-query.c

```diff
@@ -0,0 +1,71 @@
+#include "marsgaming-query.h"
+
+uint8_t
+marsgaming_query_current_profile(struct ratbag_device *device)
+{
+	uint8_t writedata[16] = { 0x02, 0x03, 0x43, 0x00, 0x01, 0x00, 0xfa, 0xfa };
+	ratbag_hidraw_set_feature_report(device, writedata[0], writedata, ARRAY_LENGTH(writedata));
+
+	uint8_t readdata[16] = { 0 };
+	ratbag_hidraw_get_feature_report(device, 0x02, readdata, ARRAY_LENGTH(readdata));
+
+	return readdata[8];
+}
+
+struct marsgaming_report_resolutions
+marsgaming_query_profile_resolutions(struct ratbag_profile *profile)
+{
+	uint8_t writedata[16] = { 0x02, 0x03, 0x4f, /* profile */ 0x00, 0x2a, 0x00, 0xfa, 0xfa } /* zeros */;
+	writedata[3] = profile->index;
```

Comment:
IMO it's more clear to set profile index directly in the initializer:
```suggestion
	uint8_t writedata[16] = {
		0x02, 0x03, 0x4f, profile->index, 0x2a, 0x00, 0xfa, 0xfa,
	};
```
Also, what does `zeros` comment mean?

Reply:
`profile->index` in `unsigned int`, so also cast it to uint8_t:
```suggestion
	uint8_t writedata[16] = {
		0x02, 0x03, 0x4f, (uint8_t)profile->index, 0x2a, 0x00, 0xfa, 0xfa,
	};
```

Reply:
I agree it is more clear. I'll change it. The `zeros` comment means that the rest of the bytes that are not explicitly initialized are set to zero. I can remove it